include/uapi/sound/sof/header.h

```diff
@@ -22,7 +25,11 @@ struct sof_abi_hdr {
 	__u32 type;		/**< component specific type */
 	__u32 size;		/**< size in bytes of data excl. this struct */
 	__u32 abi;		/**< SOF ABI version */
-	__u32 reserved[4];	/**< reserved for future use */
+	__u8 ipc_version;	/** 0: IPC3, 1: IPC4 */
+	__u8 ipc_type;		/** IPC4 typ. 0: init, 2: config, 4: large_config */
+	__u16 rsvd0[1];		/**< reserved for future use */
+	__u32 extension;	/** used in IPC4 module config/large_config */
```

Comment:
Any suggestion for the name? My concern is if we are using generic name, the code will be hard to understand. And what's more, the field size is different. It seems to hard to reuse.

Reply:
how about ipc_version_priv and ipc_version_extension or someting like that?

Reply:
IMHO, general names can be better for reuse in the future. How about
`__u8 ipc_version;	/** 0: IPC3, 1: IPC4 */`
`__u8 ipc_version_extension;	/** version-specific extension */` ... For IPC4, this field is IPC4 type. For IPC5 in the future, it can be something else.